contracts/test/contract/ethCallAware.spec.ts

```diff
@@ -38,41 +38,56 @@ describe("EthCallAware", async () => {
   const data = "0x2020";
 
   const tests = async (skipEthCallAware: boolean) => {
-    it(`allows transaction to continue (skipped: ${skipEthCallAware})`, async () => {
-      const ethCallAware = await setupEthCallAware();
-      const res = await ethCallAware.functions.testFunction(num, data, skipEthCallAware);
-      const receipt = await res.wait();
+    for (let i = 0; i < 2; i++) {
+      const opts: any = {};
+      if (!skipEthCallAware) {
+        if (i == 0) opts["gasPrice"] = ethers.BigNumber.from("0xcA11");
+        else opts["txOrigin"] = "0x000000000000000000000000000000000000cA11";
+      }
+      describe(`running tests by overloading ${i === 0 ? "gasPrice" : "txOrigin"}`, () => {
+        it(`allows transaction to continue (skipped: ${skipEthCallAware})`, async () => {
+          const ethCallAware = await setupEthCallAware();
+          const res = await ethCallAware.functions.testFunction(num, data, skipEthCallAware);
+          const receipt = await res.wait();
 
-      const event = ethCallAware.interface.parseLog(receipt.logs[0]).args as TxSuccessEvent["args"];
-      expect(event.data, "data").to.eq(data);
-      expect(event.num.toNumber(), "num").to.eq(num);
-    });
+          const event = ethCallAware.interface.parseLog(receipt.logs[0])
+            .args as TxSuccessEvent["args"];
+          expect(event.data, "data").to.eq(data);
+          expect(event.num.toNumber(), "num").to.eq(num);
+        });
 
-    it(`call reverts with data (skipped: ${skipEthCallAware})`, async () => {
-      const ethCallAware = await setupEthCallAware();
+        it(`call reverts with data (skipped: ${skipEthCallAware})`, async () => {
```

Comment:
Once you;ve got rid of skipped you could probably make `allows transaction to continue` and `call reverts with data` into a function and call it twice - once for each of the possible overrides

Reply:
just kept them as inside the loop, but could do that too